src/toil/jobStores/abstractJobStore.py

```diff
@@ -114,10 +158,118 @@ def _checkJobStoreCreation(create, exists, jobStoreString):
             raise JobStoreCreationException("The job store '%s' does not exist, so there "
                                             "is nothing to restart." % jobStoreString)
 
+    def importFile(self, srcUrl):
+        """
+        Imports the file pointed at by sourceUrl into job store. The jobStoreFileId of the new
+        file is returned.
+
+        Note that the helper method _importFile is used to read from the source and write to
+        destination (which is the current job store in this case). To implement any optimizations that
+        circumvent this, the _importFile method should be overridden by subclasses of AbstractJobStore.
+
+        Currently supported schemes are:
+            - 's3' for objects in Amazon S3
+                e.g. s3://bucket/key
+
+            - 'wasb' for blobs in Azure Blob Storage
+                e.g. wasb://container/blob
+
+            - 'file' for local files
+                e.g. file://local/file/path
+
+        :param str srcUrl: URL that points to a file or object in the storage mechanism of a
+                supported URL scheme e.g. a blob in an Azure Blob Storage container.
+        :return The jobStoreFileId of imported file.
+        :rtype: str
+        """
+        url = urlparse.urlparse(srcUrl)
+        return self._importFile(findJobStoreForUrl(url), url)
+
+    def _importFile(self, otherCls, url):
+        """
+        Refer to importFile docstring for information about this method.
+
+        :param toil.jobStores.abstractJobStore.AbstractJobStore otherCls: The class of job store to import from.
```

Comment:
I don't think you need to fully qualify the class name here.

And the type hit is incorrect because it currently specifies that an instance of AbstractJobStore is passed. I looked around and couldn't find a type hinting syntax for an argument that is a class rather than an instance of a class. A safe bet would be to use `type` as the type hint:

```
>>> class Foo(object):
...     pass
...     
>>> foo = Foo()
>>> isinstance(foo,Foo)
True
>>> isinstance(Foo,type)
True
>>> type(Foo)
<type 'type'>
>>> type(foo)
<class 'Foo'>
```